rllib/examples/checkpoints/change_config_during_training.py

```diff
@@ -0,0 +1,236 @@
+until `--stop-reward` is reached.
+
+Trial status: 1 RUNNING
+Current time: 2024-06-03 12:05:00. Total running time: 1min 0s
+Logical resource usage: 3.0/12 CPUs, 0/0 GPUs
+╭────────────────────────────────────────────────────────────────────────
+│ Trial name                    status       iter     total time (s)
+├────────────────────────────────────────────────────────────────────────
+│ PPO_CartPole-v1_7b1eb_00000   RUNNING        23            14.8372
+╰────────────────────────────────────────────────────────────────────────
+───────────────────────────────────────────────────────────────────────╮
+..._sampled_lifetime     ..._trained_lifetime     ...episodes_lifetime │
+───────────────────────────────────────────────────────────────────────┤
+              109078                   109078                      531 │
+───────────────────────────────────────────────────────────────────────╯
+
+And if you are using the `--as-test` option, you should see a finel message:
+
+```
+`env_runners/episode_return_mean` of 450.0 reached! ok
+```
+"""
+from ray.rllib.algorithms.callbacks import DefaultCallbacks
+from ray.rllib.algorithms.ppo import PPOConfig
+from ray.rllib.algorithms.algorithm_config import AlgorithmConfig
+from ray.rllib.core import DEFAULT_MODULE_ID
+from ray.rllib.examples.envs.classes.multi_agent import MultiAgentCartPole
+from ray.rllib.policy.policy import PolicySpec
+from ray.rllib.utils.metrics import (
+    ENV_RUNNER_RESULTS,
+    EPISODE_RETURN_MEAN,
+    LEARNER_RESULTS,
+)
+from ray.rllib.utils.numpy import convert_to_numpy
+from ray.rllib.utils.test_utils import (
+    add_rllib_example_script_args,
+    check,
+    run_rllib_example_script_experiment,
+)
+from ray.tune.registry import register_env
+
+
+parser = add_rllib_example_script_args(
+    default_reward=450.0, default_timesteps=10000000, default_iters=2000
+)
+parser.add_argument(
+    "--stop-reward-first-config",
+    type=float,
+    default=150.0,
+    help="Mean episode return after which the Algorithm on the first config should "
+    "stop training.",
+)
+# By default, set `args.checkpoint_freq` to 1 and `args.checkpoint_at_end` to True.
+parser.set_defaults(
+    enable_new_api_stack=True,
+    checkpoint_freq=1,
+    checkpoint_at_end=True,
+)
+
+
+if __name__ == "__main__":
+    args = parser.parse_args()
+
+    register_env(
+        "ma_cart", lambda cfg: MultiAgentCartPole({"num_agents": args.num_agents})
+    )
+
+    # Simple generic config.
+    base_config = (
+        PPOConfig()
+        .environment("CartPole-v1" if args.num_agents == 0 else "ma_cart")
+        .training(lr=0.0001)
+    )
+
+    # Setup multi-agent, if required.
+    if args.num_agents > 0:
+        base_config.multi_agent(
+            policies={
+                f"p{aid}": PolicySpec(
+                    config=AlgorithmConfig.overrides(
+                        lr=5e-5
+                        * (aid + 1),  # agent 1 has double the learning rate as 0.
```

Comment:
Good catch. I think, then this example breaks :( . I'll just force the setting to always be 2 in this particular example script, otherwise produce an error. :)